### PR TITLE
Revert 'Save Draft' changes

### PR DESCRIPTION
**Pull request by kean**
We've received some feedback about the "Save Draft" changes in 24.9, and decide to revert a part of the change.

Changes:

- Add "Save Draft", "Discard Changes" dialog when you tap "Back" (as opposed to saving automatically)
- Add "Save Draft" more menu action for saving without as you edit (folks asked for it and it's more discoverable – let's just keep it for now)

## To test:

Test matrix: Gutenberg and Aztec.

**Scenario 1.1**

- Tap "+" to create a new draft
- Do **NOT** make any changes
- Tap "Back"
- ✅ Verify that the editor was closed with no confirmation and the draft post is not saved

**Scenario 1.2**

- Tap "+" to create a new draft
- Make some changes
- Tap "Back"
- ✅ Verify that the confirmation dialog appeared
  - ✅ Verify that if you save "Save Draft", it saves a new post
  - ✅ Verify that if you save "Discard Draft", it discards the post
 
**Scenario 1.3**

- Open an existing draft
- Make some changes
- Tap "Back"
- ✅ Verify that the confirmation dialog appeared
  - ✅ Verify that if you save "Save Draft", it saves a new post
  - ✅ Verify that if you save "Discard Changes", it discards the recent changes (and only them)

**Scenario 2.1**

- Tap "+" to create a new draft
- Make some changes
- Tap "More" / "Save Draft"
- ✅ Verify that it saves the draft
- Open "More" / "Post Settings"
- ✅ Verify that the slug was populated

**Scenario 2.2**

- Open an existing draft
- Do NOT make any changes
- Tap "More" / "Save Draft"
- ✅ Verify that the "Save Draft" option is grayed out
- ❗ known issue: there is a bit of a delay after modifying content and before the editor "learns" about the changes

## Regression Notes
1. Potential unintended areas of impact: Post Editor
2. What I did to test those areas of impact (or what existing automated tests I relied on): n/a
3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/291">25.0 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by dvdchr — APPROVED** [review]
Tested and verified that these scenarios worked well:

- ✅  Scenario 1.1, 1.2, 1.3
- ✅  Scenario 2.1, 2.2

The changes look good to me, although I wanted to note the new localized strings. It seems like `classicEditor.moreMenu.saveDraft` and `postEditor.moreMenu.saveDraft` are new strings, and we _might_ not have enough time to translate them in time for 25.0.

IIRC translation takes around 5 days for mag16 but we might be able to ask this to be expedited. In case we don't have enough time, I would suggest 2 options:

1. Temporarily reuse `postEditor.closeConfirmationAlert.saveDraft`. Although not ideal due to the semantic mismatch, this key has been translated. We can use this just for 25.0, and assign the new string keys targeting `trunk` later.
2. Go as is with the new keys, and manually update the entries in translate.wordpress.org; copying the translation values from `postEditor.closeConfirmationAlert.saveDraft`.

I'm approving this PR to unblock and will leave the above to your decision. 🙂

**Comment by kean** [comment]
Hey, great catch with localization. I haven't thought about that. I added a deprecated API and reused the value as you suggested:

```
extension PostEditorAction {
    /// This value needs to be replaced with a separate entry in l10n.
    /// - warning: Deprecated (kahu-offline-mode)
    static var saveDraftLocalizedTitle: String { Strings.closeConfirmationAlertSaveDraft }
}
```

I'm in the process of removing the existing deprecations in trunk, so I'll take care of it once it gets merged back.

**Pull request merged**
